crud_handler/crud_handler.go

```diff
@@ -0,0 +1,191 @@
+package crud_handler
+
+import (
+	"encoding/json"
+	"fmt"
+	"main/transformer"
+	"net/http"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/google/uuid"
+)
+
+type TransformRequest struct {
+	Type        string `json:"type"`
+	CaesarShift int    `json:"shift,omitempty"`
+	Input       string `json:"input,omitempty"`
+}
+
+type TransformResult struct {
+	Id          string `json:"uuid"`
+	Type        string `json:"type"`
+	CaesarShift int    `json:"shift,omitempty"`
+	Result      string `json:"result,omitempty"`
+	Created_at  int64  `json:"created"`
+	Updated_at  int64  `json:"updated,omitempty"`
+}
+
+// Fake database just for example
+var temp1, temp2, temp3 = uuid.NewString(), uuid.NewString(), uuid.NewString()
+var TransformRequests = map[string]*TransformRequest{
+	temp1: &TransformRequest{Type: "reverse", Input: "123456789"},
+	temp2: &TransformRequest{Type: "base64", Input: "Man"},
+	temp3: &TransformRequest{Type: "caesar", CaesarShift: 1, Input: "zab"},
+}
+var TransformResults = map[string]*TransformResult{
+	temp1: &TransformResult{Id: temp1, Type: "reverse", Result: "987654321", Created_at: time.Now().Unix()},
+	temp2: &TransformResult{Id: temp2, Type: "base64", Result: "TWFu", Created_at: time.Now().Unix() + 1},
+	temp3: &TransformResult{Id: temp3, Type: "caesar", CaesarShift: 1, Result: "abc", Created_at: time.Now().Unix() + 2},
+}
+
+func SetJSONContentType(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Add("Content-Type", "application/json")
+		next.ServeHTTP(w, r)
+	})
+}
+
+func HandlePost(w http.ResponseWriter, r *http.Request) {
+	request := new(TransformRequest)
+	dec := json.NewDecoder(r.Body)
+	err := dec.Decode(&request)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	if request.Type != "reverse" && request.Type != "caesar" && request.Type != "base64" {
+		http.Error(w, "expected tranformation type field: reverse/caesar/base64", http.StatusInternalServerError)
+		return
+	}
+	if request.Type == "caesar" && request.CaesarShift == 0 {
+		http.Error(w, "expected shift field (not 0)", http.StatusInternalServerError)
+		return
+	}
+	if request.Input == "" {
+		http.Error(w, "expected input field", http.StatusInternalServerError)
+		return
+	}
+	fmt.Fprintln(w, "Status ", http.StatusCreated, ": ", http.StatusText(http.StatusCreated))
+
+	result := new(TransformResult)
+	result.Id = uuid.NewString()
```

Comment:
It should be panic here, because result is `nil`.
Just init result as following

```go
result := TransformResult{
  ID: uuid.New(),
  Type: request.Type,
  ...
}
```